relengapi/blueprints/slaveloan/tasks.py

```diff
@@ -258,12 +258,43 @@ def waitfor_action(self, action_id, loanid):
         self.retry(exc=exc)
 
 
+@task(bind=True, max_retries=None)
+@add_to_history(
+    before="Calling slaveapi's disable method to disable from buildbot",
+    after="Disable request sent")
+def start_disable_slave(self, machine, loanid):
+    try:
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url.path.add(machine).add("actions").add("shutdown_buildslave")
+        ret = retry(requests.post, args=(str(url),), ).json()
+        return (ret["requestid"], machine)
+    except Exception as exc:
+        logger.exception(exc)
+        self.retry(exc=exc)
+
+
+@task(bind=True, max_retries=None)
+@add_to_history(
+    after="Noticed that machine was disabled (or waiting timed out)")
+def waitfor_disable_slave(self, data, loanid):
+    requestid, machine = data
+    try:
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url = furl(current_app.config.get("SLAVEAPI_URL", None))
+        url.path.add(machine).add("actions").add("shutdown_buildslave")
```

Comment:
OK, so its naming is worse than you indicated: not only is it not shutting
down the host, it's not shutting down the buildslave process, or indeed an
action of any sort.

On Tue, Mar 31, 2015 at 12:31 PM, Justin Wood (Callek) <
notifications@github.com> wrote:

> In relengapi/blueprints/slaveloan/tasks.py
> <https://github.com/mozilla/build-relengapi-slaveloan/pull/31#discussion_r27496308>
> :
>
> > +        ret = retry(requests.post, args=(str(url),), ).json()
> > +        return (ret["requestid"], machine)
> > +    except Exception as exc:
> > +        logger.exception(exc)
> > +        self.retry(exc=exc)
> > +
> > +
> > +@task(bind=True, max_retries=None)
> > +@add_to_history(
> > +    after="Noticed that machine was disabled (or waiting timed out)")
> > +def waitfor_disable_slave(self, data, loanid):
> > +    requestid, machine = data
> > +    try:
> > +        url = furl(current_app.config.get("SLAVEAPI_URL", None))
> > +        url = furl(current_app.config.get("SLAVEAPI_URL", None))
> > +        url.path.add(machine).add("actions").add("shutdown_buildslave")
>
> My mental model here is:
> waitfor_
> --> POLL slaveapi to check status on thing.
>
> this GET request is not "doing" something per-se, just checking status
> using an external data source
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mozilla/build-relengapi-slaveloan/pull/31/files#r27496308>
> .
>